src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    const sideSorted: number[] = [a, b, c].sort(
+      (side1, side2) => side2 - side1,
+    );
+
+    if (sideSorted[0] >= sideSorted[1] + sideSorted[2]) {
+      throw new Error(`Sides can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const square = Math.sqrt(p * ((p - this.a) * (p - this.b) * (p - this.c)));
+
+    return Math.floor(square * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape: string = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
The variable `radius` should be accessed using `this.radius` within the constructor to refer to the class property.